File: solutions/cpp/lasagna-master/1/lasagna_master.cpp
#include "lasagna_master.h"
#include <iostream>
namespace lasagna_master {

    int preparationTime(const vector<string>& layers, int avg_time){
        return layers.size() * avg_time;
    }

    amount quantities(const vector<string>& layers){
        amount ingreds{0, 0.0};
        
        for(const auto& layer : layers){
            if(layer == "noodles"){
                ingreds.noodles += 50;
            }else if(layer == "sauce"){
                ingreds.sauce += 0.2;
            }
        }

        return ingreds;
    }

    void addSecretIngredient(vector<string>& ownRecipe, const vector<string>& friendRecipe){
        ownRecipe[ownRecipe.size() - 1] = friendRecipe[friendRecipe.size() - 1];
    }

    vector<double>scaleRecipe(const vector<double>& quantities, const int& portions){
        vector<double> newQuantities{};

        for(const auto& quantity : quantities){
            newQuantities.push_back(quantity * portions / 2);
        }

        return newQuantities;
    }

    void addSecretIngredient(vector<string>& ownRecipe, const string& secretIngredient){
         ownRecipe[ownRecipe.size() - 1] = secretIngredient;
    }
    
}  // namespace lasagna_master

File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/**
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 *
 * export function yourFunction(...) {
 *   ...
 * }
 */

export function cookingStatus(remainingTime){
  switch(remainingTime){
    case 0:
      return 'Lasagna is done.';
    case undefined:
      return 'You forgot to set the timer.';
    default:
      return 'Not done, please wait.'
  }
}

export function preparationTime(layers, prepTime=2){
  return layers.length * prepTime;
}

const noodle = 50;
var sauce = 0.2;

export function quantities(layers){
  var nQ = 0;
  var sQ = 0;
  
  for(var i = 0; i < layers.length; i++){

    if(layers[i] == 'noodles'){
      nQ++;
    } else if(layers[i] == 'sauce'){
      sQ++;
    }
  } 

  return{
    noodles: nQ * noodle,
    sauce: sQ * sauce,
  };
}

export function addSecretIngredient(friendList, myList){
  myList.push(friendList[friendList.length - 1])
}

export function scaleRecipe(recipe, portion=2){
  const scale = portion / 2;
  var newRecipe = {};
  
  for(var key in recipe){
    newRecipe[key] = recipe[key] * scale;
  }
  
  return newRecipe;
}